packages/cli/commands/tools/initial-analysis.ts

```diff
@@ -0,0 +1,226 @@
+import { AnalysisType } from "@replayio/data/src/analysis/dependencyGraphShared";
+import { AnalysisInput } from "@replayio/data/src/analysis/dgSpecs";
+import { runAnalysis } from "@replayio/data/src/analysis/runAnalysis";
+import { ExecutionDataAnalysisResult } from "@replayio/data/src/analysis/specs/executionPoint";
+import { scanGitUrl } from "@replayio/data/src/gitUtil/gitStringUtil";
+import LocalGitRepo from "@replayio/data/src/gitUtil/LocalGitRepo";
+import { InspectPointResult } from "@replayio/data/src/recordingData/types";
+import ReplaySession, {
+  getOrCreateReplaySession,
+} from "@replayio/data/src/recordingData/ReplaySession";
+import {
+  scanAnnotationDataUrl,
+  scanReplayUrl,
+} from "@replayio/data/src/recordingData/replayStringUtil";
+import { assert } from "@replayio/data/src/util/assert";
+import { program } from "commander";
+import createDebug from "debug";
+
+import { printCommandError, printCommandResult } from "../../commandsShared/commandOutput";
+import { AnalysisToolMetadata } from "./tools-shared";
+
+const debug = createDebug("replay:initial-analysis");
+
+type LegacyOptions = {
+  prompt: string;
+  workspacePath?: string;
+  isWorkspaceRepoPath?: boolean;
+  forceDelete?: boolean;
+};
+
+export type InitialAnalysisCommandOptions = {
+  prompt: string;
+};
+
+export interface InitialAnalysisResult extends InspectPointResult {
+  thisPoint: string;
+  commentText?: string;
+  reactComponentName?: string;
+  consoleError?: string;
+  metadata: AnalysisToolMetadata;
+}
+
+const NewRecordingIds = ["011f1663-6205-4484-b468-5ec471dc5a31", "7dfc5103-8060-4128-a1b6-20d0a56aadcc"];
+function shouldUseLegacyMode(recordingId: string | undefined): boolean {
+  if (!recordingId) {
+    // If we have no recordingId from the problem description, default to new mode.
+    return false;
+  }
+
+  // If the recordingId is in the array, return false (new mode), else true (legacy mode)
+  return !NewRecordingIds.includes(recordingId);
+}
+
+program
+  .command("initial-analysis")
+  .description(
+    "Perform initial analysis of a recording. Legacy logic for annotation execution points may run if checkLegacy() returns true."
+  )
+  .option(
+    "-p, --prompt <prompt>",
+    "Prompt text, containing recordingId and maybe other relevant data sources."
+  )
+
+  // Legacy options (ignored when not legacy).
+  .option("-w, --workspacePath <workspacePath>", "Local file path of the workspace.", false)
+  .option(
+    "-i, --isWorkspaceRepoPath",
+    "If set, `workspacePath` is path to the repo. Otherwise, it's a parent path.",
+    false
+  )
+  .option("-f, --forceDelete", "Delete the target repo directory if it already exists.", false)
+  
+  // Action handler.
+  .action(async (options: InitialAnalysisCommandOptions & LegacyOptions) => {
+    // We need a recordingId either from the legacy prompt or from the new mode.
+    const { recordingId } = scanReplayUrl(options.prompt);
+
+    if (shouldUseLegacyMode(recordingId)) {
```

Comment:
Only use the new workflow on selected recordings for now.
We generally want to remove this check once we can confirm that it works in all previous samples.